mods/cnc/maps/nod07b/nod07b.lua

```diff
@@ -0,0 +1,265 @@
+
+CaptureStructures = function(actor)
+	for i = 1, #WhitelistedStructures do
+		structures = player.GetActorsByType(WhitelistedStructures[i])
+		if #structures > 0 then
+			if not actor.IsDead and not structures[1].IsDead then
+				actor.Capture(structures[1])
+				return
+			end
+		end
+	end
+end
+
+CheckForSams = function()
+	local sams = player.GetActorsByType("sam")
+	return #sams >= 3
+end
+
+searches = 0
+getAirstrikeTarget = function()
+	local list = player.GetGroundAttackers()
+
+	if #list == 0 then
+		return
+	end
+
+	local target = list[DateTime.GameTime % #list + 1].CenterPosition
+
+	local sams = Map.ActorsInCircle(target, WDist.New(8 * 1024), function(actor)
+		return actor.Type == "sam" end)
+
+	if #sams == 0 then
+		searches = 0
+		return target
+	elseif searches < 6 then
+		searches = searches + 1
+		return getAirstrikeTarget()
+	else
+		searches = 0
+		return nil
+	end
+end
+
+GetCargo = function(team)
+	cargo = { }
+	for type, count in pairs(team.units) do
+		for i = 1, count, 1 do
+			cargo[#cargo + 1] = type
+		end
+	end
+	return cargo
+end
+
+InsertNodUnits = function()
+	Media.PlaySpeechNotification(player, "Reinforce")
+	Reinforcements.Reinforce(player, NodUnitsBikes, { ReinforcementsSpawnLeft.Location, ReinforcementsBikesRally.Location }, 1)
+	Reinforcements.Reinforce(player, NodUnitsBuggys, { ReinforcementsSpawnRight.Location, ReinforcementsBuggyRally.Location }, 50)
+	Reinforcements.Reinforce(player, NodUnitsGunners, { ReinforcementsSpawnLeft.Location, ReinforcementsGunnersRally.Location }, 50)
+	Reinforcements.Reinforce(player, NodUnitsRockets, { ReinforcementsSpawnRight.Location, ReinforcementsRocketsRally.Location }, 50)
+	Trigger.AfterDelay(DateTime.Seconds(6), function()
+		Reinforcements.Reinforce(player, { 'mcv' }, { ReinforcementsSpawnCenter.Location, ReinforcementsMCVRally.Location })
+		Reinforcements.Reinforce(player, NodUnitsTanks, { ReinforcementsSpawnCenter.Location, ReinforcementsTanksRally.Location }, 50)
+	end)
+end
+
+SendAttackWave = function(team)
+	for type, amount in pairs(team.units) do
+		count = 0
+		actors =  enemy.GetActorsByType(type)
+		Utils.Do(actors, function(actor)
+			if actor.IsIdle and count < amount then
+				SetAttackWaypoints(actor, team.waypoints)
+				if actor.Type == "e6" then 
+					CaptureStructures(actor)
+				else
+					IdleHunt(actor)
+				end
+				count = count + 1
+			end
+		end)
+	end
+end
+
+SetAttackWaypoints = function(actor, waypoints)
+	if not actor.IsDead then
+		Utils.Do(waypoints, function(waypoint)
+			actor.AttackMove(waypoint.Location)
+		end)
+	end
+end
+
+SendGDIAirstrike = function(hq, delay)
+	if not hq.IsDead and hq.Owner == enemy then
```

Comment:
Wouldn't that prevent airstrikes on the entire map? I thought the point of that code was to secure your base from airstrikes.